### DISCUSSION TITLE
Progress bars: printing iterations/second and unbounded processes

### DISCUSSION THREAD

**Discussion by wbthomason** [Textualize/rich#926] [Q&A]
Hi! I'm wondering if there's a way to use `rich.progress` to print information - specifically the number of iterations per second - for a process with an unknown number of steps. With `tqdm`, I do this as follows:
```python
from tqdm import tqdm
progress = tqdm(desc='My process', unit='')
while not process_is_done():
  progress.update()
```

Is there a way to do this or something similar with `rich`? Apologies if this is a question already answered in the docs - I read through what seemed to be the relevant sections, but may have missed something.

**Comment by willmcgugan** [comment]
Yes, tasks have a `speed` attribute which is iterations per second.

Have a look at [TransferSpeedColumn](https://github.com/willmcgugan/rich/blob/master/rich/progress.py#L405), which assumes your units are bytes of data, but you could modify that to display iterations per second.

Probably worth adding that to the core lib, so feel free to PR that.

**Reply by wbthomason** [reply]
Thanks! I'll try to put together a PR once I get things working.

Also, looking at https://github.com/willmcgugan/rich/blob/0e8f4747cae99d9ccb2596c966af5737e955a932/rich/progress.py#L440, I'm guessing that there is no way to specify a task with an unknown total bound, correct? (I assume that field would need to be `Optional[float]` rather than `float` if it was possible)

**Comment by aziz0x00** [comment]
Hey there, is this problem of "number of iterations per second" is resolved or a PR for it is ok?

**Comment by deo-nikita** [comment]
I'd love this feature too

**Reply by deo-nikita** [reply]
so I can estimate the speed (iterations per second) at which my deep learning model works, not only % finished and time remaining

**Comment by NichtJens** [comment]
@aziz0x00 and @deo-nikita Since I had a similar usecase, I just submitted #3533. Hope it will help you too (if/once it's accepted).

**Comment by Ar-Kareem** [comment]
I would also prefer an option to provide a string other than "bytes". Even just "iterations" would suffice.
